httpclient/port.go

```diff
@@ -0,0 +1,20 @@
+package httpclient
+
+import (
+	"net"
+	"strconv"
+)
+
+func FindFreePort() (string, error) {
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+	if err != nil {
+		return "", err
+	}
+
+	listener, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		return "", err
+	}
+
+	return strconv.Itoa(listener.Addr().(*net.TCPAddr).Port), nil
```

Comment:
With the new data flow, the listener is not reused by `http.Server(listener)` but instead a new one is created via `http.Server.ListenAndServe(port)`.

We should then close this one here when we're done with it.

```suggestion
	listener, err := net.ListenTCP("tcp", addr)
	if err != nil {
		return "", err
	}
	defer listener.Close()

	return strconv.Itoa(listener.Addr().(*net.TCPAddr).Port), nil
```